### PR TITLE
Add support for getting / replacing topics (#634)

### PR DESCRIPTION
**Pull request by wyardley**
This is mostly @peterkline's work, just trying to replace #635 (which hasn't been updated) and adjust for some of the changes requested / discussed there.

This seems to resolves the existing test failure, and I think resolves most of the other feedback in the other PR. It also switches the input to `replace_topics()` to the expected array as discussed. Would appreciate feedback, esp. on the test case I added. Because there's no expected return from the method, I think that test is acceptable as-is? I'm not that familiar with this module's test setup, though, so let me know if I should update anything.

I didn't have much luck rebasing, so I just patched, but I did retain original author on the first commit.
cc @sfdye

### CODE REVIEW

File: github/tests/Repository.py [hunk 1]
```diff
@@ -711,6 +712,13 @@ def testStatistics(self):
     def testGetLicense(self):
         self.assertEqual(len(self.repo.get_license().content), 47646)
 
+    def testGetTopics(self):
+        topic_list = self.repo.get_topics()
+        topic = u'github'
+        self.assertEquals(topic in topic_list, True)
+
+    def testReplaceTopics(self):
+        self.repo.replace_topics(['github', 'testing'])
```
Comment (wyardley) [inline github/tests/Repository.py]: Looking at some of the other tests, I think no assertion is needed here? Let me know if this needs to be improved.

File: github/Repository.py [hunk 2]
```diff
@@ -2426,6 +2446,22 @@ def protect_branch(self, branch, enabled, enforcement_level=github.GithubObject.
             input=post_parameters
         )
 
+    def replace_topics(self, topics):
+        """
+        :calls: `PUT /repos/:owner/:repo/topics <http://developer.github.com/v3/repos>`_
```
Comment (sfdye) [inline github/Repository.py]: https://developer.github.com/v3/repos/#replace-all-topics-for-a-repository

File: github/Repository.py [hunk 3]
```diff
@@ -2304,6 +2312,18 @@ def get_teams(self):
             None
         )
 
+    def get_topics(self):
+        """
+        :calls: `GET /repos/:owner/:repo/topics <http://developer.github.com/v3/repos>`_
```
Comment (sfdye) [inline github/Repository.py]: Let's add the exact link:
https://developer.github.com/v3/repos/#list-all-topics-for-a-repository

File: github/tests/Repository.py [hunk 4]
```diff
@@ -711,6 +712,13 @@ def testStatistics(self):
     def testGetLicense(self):
         self.assertEqual(len(self.repo.get_license().content), 47646)
 
+    def testGetTopics(self):
+        topic_list = self.repo.get_topics()
+        topic = u'github'
+        self.assertEquals(topic in topic_list, True)
```
Comment (sfdye) [inline github/tests/Repository.py]: `self.assertIn`?

File: github/tests/Repository.py (hunk 1 above)
Comment (sfdye) [reply github/tests/Repository.py]: It's fine

File: github/Repository.py [hunk 5]
```diff
@@ -670,6 +670,14 @@ def teams_url(self):
         self._completeIfNotSet(self._teams_url)
         return self._teams_url.value
 
+    @property
+    def topics(self):
+        """
+        :type: list
```
Comment (sfdye) [inline github/Repository.py]: list of string

**Comment by wyardley** [comment]
@sfdye Thanks for the review feedback, I think I addressed your points -- let me know how that looks.

**Comment by sfdye** [comment]
@wyardley Thank you very much for putting this together. Merged!

**Pull request merged**